Gifski/AppDelegate.swift

```diff
@@ -117,11 +112,15 @@ final class AppDelegate: NSObject, NSApplicationDelegate {
 		return error
 	}
 
-	private func convertVideoUrl() {
-		if let videoUrl = urlToConvertOnLaunch, hasFinishedLaunching {
-			mainWindowController.convert(videoUrl)
+	private func runFinishedLaunchingCompletions() {
+		for completion in finishedLaunchingCompletions {
+			completion()
 		}
 	}
+
+	private func onFinishedLaunching(_ completion: @escaping () -> Void) {
+		finishedLaunchingCompletions.append(completion)
+	}
```

Comment:
This is closer to what I had in mind, but you could move it to a separate struct (with static methods) that handles this logic. You could even make it more self contained by listening to the `didFinishLaunchingNotification` notification.

Don't forget to clear `finishedLaunchingCompletions` when all the functions have run so not to leak.

Reply:
I changed it as you said. It is now nearly self contained. But to make sure I also run the closures if the app was already started before any completions were added, I have to tell it once the application finished launching. That's what I did.    
In all other cases I am using notificationcenter.